index.js

```diff
@@ -27,10 +27,15 @@ const HEADER = `// Type definitions for WebExtension Development in FireFox ${ar
 // TypeScript Version: 2.4
 // Generated using script at github.com/bomjacob/definitelytyped-firefox-webext-browser
 
-interface WebExtEventListener<T extends (...args: any[]) => any> {
-    addListener: (callback: T) => void;
-    removeListener: (listener: T) => void;
-    hasListener: (listener: T) => boolean;
+interface WebExtEventBase<TAddListener, TCallback> {
+    addListener: TAddListener;
+
+    removeListener (cb: TCallback): void;
+
+    hasListener (cb: TCallback): boolean;
+}
+
+interface WebExtEvent<TCallback extends (...args: any[]) => any> extends WebExtEventBase<(callback: TCallback) => void, TCallback> {
```

Comment:
like the other interface, this could be replaced with:

```typescript
type WebExtEvent<TCallback extends (...args: any[]) => any> = WebExtEventBase<(callback: TCallback) => void, TCallback>;
```